types/lodash/common/lang.d.ts

```diff
@@ -574,7 +574,7 @@ declare module "../index" {
         isEmpty<T extends { __trapAny: any }>(value?: T): boolean;
         isEmpty(value: string): value is '';
         isEmpty(value: Map<any, any> | Set<any> | List<any> | null | undefined): boolean;
-        isEmpty<T extends object>(value: T | null | undefined): value is EmptyObjectOf<T> | null | undefined;
+        isEmpty<T extends object>(value: EmptyObjectOf<T> | null | undefined): value is EmptyObjectOf<T> | null | undefined;
```

Comment:
I don't think this is what we want since it effectively reduces to `isEmpty(value: {} | null | undefined): true`.

I'm not sure what your issue is, but it will likely be fixed by #62409, can you verify?

Reply:
I'm encountered with building errors which piont to the code.

```text
ERROR in node_modules/@types/lodash/common/lang.d.ts(577,74): error TS2677: A type predicate's type must be assignable to its parameter's type.
  Type 'EmptyObjectOf<T>' is not assignable to type 'T'.
    Type 'EmptyObject<T>' is not assignable to type 'T'.
```

This commit could fix this problem.

And, #62409 does not fix this problem.